File: liqp-core/src/main/java/liqp/filters/Prepend.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Prepend extends Filter {

    /*
     * (Object) append(input, string)
     *
     * add one string to another
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        return super.asString(super.get(0, params)) + super.asString(value);
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Prepend.kt
package liqp.filter

import liqp.context.LContext

class Prepend : LFilter() {

  /**
   * (Object) append(input, string)
   *
   * add one string to another
   */
  override fun onFilterAction(params: FilterParams, value: Any?, context: LContext): Any? {
    context.run {
      val content = asString(value) ?: ""
      val prepend: String = asString(params[0]) ?: return content
      return prepend + content
    }
  }
}
